jax/experimental/jax2tf/tfjs_demo/Sketcher/sketcher.py

```diff
@@ -0,0 +1,278 @@
+      self.feature_extractor = FlaxQuickDraw.FeatureExtractorModule()
+
+    @nn.compact
+    def __call__(self, x):
+      x = self.feature_extractor(x)
+      x = nn.Dense(features=6)(x) # 100
+      x = nn.softmax(x)
+      return x
+
+  @staticmethod
+  def predict(params, inputs):
+    """A functional interface to the trained Module."""
+    return FlaxQuickDraw.Module().apply({'params': params}, inputs)
+
+  @staticmethod
+  #@jax.vmap
+  def loss(params, inputs, labels):
+    predictions = FlaxQuickDraw.predict(params, inputs)
+    onehot_labels = common_utils.onehot(labels, predictions.shape[-1])
+    loss = -jnp.sum(onehot_labels * nn.log_softmax(predictions), axis=-1)
+    normalizing_factor = onehot_labels.sum()
+
+    return loss.sum() / normalizing_factor
+
+  @staticmethod
+  def update(optimizer, batch_inputs, batch_labels):
+    def batch_loss(model, batch_inputs, batch_labels):
+      return jnp.mean(FlaxQuickDraw.loss(model, batch_inputs, batch_labels))
+    grad = jax.grad(batch_loss)(optimizer.target, batch_inputs, batch_labels)
+    optimizer = optimizer.apply_gradient(grad)
+    return optimizer
+
+  @staticmethod
+  def accuracy(predict: Callable, params, dataset):
+    def top_k(x, k):
+      bcast_idxs = jnp.broadcast_to(np.arange(x.shape[-1]), x.shape)
+      sorted_vals, sorted_idxs = lax.sort_key_val(x, bcast_idxs)
+      topk_vals = (
+          lax.slice_in_dim(sorted_vals, -k, sorted_vals.shape[-1], axis=-1))
+      topk_idxs = (
+          lax.slice_in_dim(sorted_idxs, -k, sorted_idxs.shape[-1], axis=-1))
+      return topk_vals, topk_idxs
+    def _per_batch(inputs, labels):
+      predictions = predict(params, inputs)
+      return jnp.mean(top_k(predictions, 1)[1] == labels)
+    batched = [_per_batch(inputs, labels) for inputs, labels in dataset]
+    return jnp.mean(jnp.stack(batched))
+
+  @staticmethod
+  def train(dir_path, classes):
+    """Trains a Flax QuickDraw predictor."""
+    train_ds, test_ds = load_classes(dir_path, classes)
+    rng = jax.random.PRNGKey(0)
+
+    init_shape = jnp.ones((1, 28, 28, 1), jnp.float32)
+    initial_params = FlaxQuickDraw.Module().init(rng, init_shape)["params"]
+    optimizer = flax.optim.Adam(
+        learning_rate=0.001, beta1=0.9, beta2=0.999).create(initial_params)
+    for epoch in range(5):
+      start_time = time.time()
+      for inputs, labels in train_ds:
+        optimizer = jax.jit(FlaxQuickDraw.update)(optimizer, inputs, labels)
+      epoch_time = time.time() - start_time
+      train_acc = FlaxQuickDraw.accuracy(
+          FlaxQuickDraw.predict, optimizer.target, train_ds)
+      test_acc = FlaxQuickDraw.accuracy(
+          FlaxQuickDraw.predict, optimizer.target, test_ds)
+      print("Epoch {} in {:0.2f} sec".format(epoch, epoch_time))
+      print("Training set accuracy {}".format(train_acc))
+      print("Test set accuracy {}".format(test_acc))
+
+    return (FlaxQuickDraw.predict, optimizer.target,
+            FlaxQuickDraw.feature_extractor,
```

Comment:
We don't need the feature extractor and its params, let's inline the feature extractor in the model.